### PR TITLE
Make port obsolete

### LINKED ISSUE #262: hardcoded port for https?
using 2.0 beta

```
if [type] == "syslog" {
                        elasticsearch {
                                hosts => ["w530:9200","w530:9201","w530:9202"]
                                index => "logstash-syslog-%{+YYYY.MM.dd}"
                                user => "logstash"
                                password => "secret"
				ssl => true
				keystore => "/opt/elk/PRODSEC/FS/secure/node1.jks"
				keystore_password => "secret"
				truststore => "/opt/elk/PRODSEC/FS/secure/ca-truststore.jks"
				truststore_password => "secret"
                        }
                }
```

```
The error reported is: 
  the scheme https does not accept registry part: w530:9200:9200 (or bad hostname?)
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/jruby/lib/ruby/1.9/uri/generic.rb:214:in `initialize'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/jruby/lib/ruby/1.9/uri/http.rb:84:in `initialize'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/jruby/lib/ruby/1.9/uri/common.rb:214:in `parse'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/jruby/lib/ruby/1.9/uri/common.rb:747:in `parse'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.13/lib/elasticsearch/transport/client.rb:149:in `__extract_hosts'
org/jruby/RubyArray.java:2414:in `map'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.13/lib/elasticsearch/transport/client.rb:145:in `__extract_hosts'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.13/lib/elasticsearch/transport/client.rb:109:in `initialize'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.13/lib/elasticsearch/transport.rb:26:in `new'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.0.0.beta6-java/lib/logstash/outputs/elasticsearch/http_client.rb:109:in `build_client'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.0.0.beta6-java/lib/logstash/outputs/elasticsearch/http_client.rb:17:in `initialize'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.0.0.beta6-java/lib/logstash/outputs/elasticsearch.rb:256:in `register'
org/jruby/RubyArray.java:1613:in `each'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0.beta1-java/lib/logstash/pipeline.rb:164:in `start_outputs'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0.beta1-java/lib/logstash/pipeline.rb:83:in `run'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0.beta1-java/lib/logstash/agent.rb:155:in `execute'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0.beta1-java/lib/logstash/runner.rb:90:in `run'
org/jruby/RubyProc.java:271:in `call'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0.beta1-java/lib/logstash/runner.rb:95:in `run'
org/jruby/RubyProc.java:271:in `call'
/opt/elk/PRODSEC/logstash/logstash-2.0.0-beta1/vendor/bundle/jruby/1.9/gems/stud-0.0.21/lib/stud/task.rb:12:in `initialize'
```

### PR DESCRIPTION
**Pull request by untergeek**
- Update flag and docs
 - Remove references from code
 - Make sure default port 9200 is added if no port specified
 - Add tests to validate changes

The Atom editor cleaned up a bunch of trailing white space issues.  Sorry about that.

fixes #262

### CODE REVIEW

**Comment by untergeek** [comment]
New tests pass:

```
    with a port other than 9200 specified
      should properly set the specified port on the HTTP client
    without a port specified
      should properly set the default port (9200) on the HTTP client
```

**Comment by suyograo** [comment]
LGTM

File: lib/logstash/outputs/elasticsearch/http_client.rb [hunk 1]
```diff
@@ -83,12 +86,16 @@ def sniff!
 
     def build_client(options)
       hosts = options[:hosts] || ["127.0.0.1"]
-      port = options[:port] || 9200
       client_settings = options[:client_settings] || {}
 
       uris = hosts.map do |host|
         proto = client_settings[:ssl] ? "https"  : "http"
-        "#{proto}://#{host}:#{port}#{client_settings[:path]}"
+        if !(host =~ /:/).nil?
```
Comment (ph) [inline lib/logstash/outputs/elasticsearch/http_client.rb]: This would work and should be a bit more strict?

```ruby
if host =~ /:\d+\z/
```

**Comment by ph** [comment]
Added a small comment on regular expression, other than that LGTM

**Pull request closed without merging**